attestation-agent/attester/src/se/mod.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright (C) Copyright IBM Corp. 2024
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use super::Attester;
+use anyhow::*;
+use log::debug;
+use pv::{
+    request::BootHdrTags,
+    uv::{AttestationCmd, ConfigUid, UvDevice},
+};
+use serde::{Deserialize, Serialize};
+use serde_json;
+use serde_with::{base64::Base64, serde_as};
+
+pub fn detect_platform() -> bool {
+    // run always on s390x machine
+    let v = std::fs::read("/sys/firmware/uv/prot_virt_guest").unwrap_or_else(|_| vec![0]);
```

Comment:
Why not just return false in the else case here?

And for the next step, can you compare the string directly rather than converting it to an int.

Also if we need the variable v, can we rename it to something more descriptive?

Reply:
Using an int completely makes sense.
He took the code from [pv_guest_bit_set](https://docs.rs/s390_pv/0.10.0/s390_pv/misc/fn.pv_guest_bit_set.html) in the `s390_pv` crate.
-> just use the exported function from that crate (you use anyways for the attestation flow).
 
 This also describes better what you are doing:
 that sysfs entry (if it exists) just says if its a SE-guest (not backed by attestation) and can be used as an early exit for e.g. an attestation flow.

Reply:
Hopefully, @fitzthum you agree with what @steffen-eiden explained, thank you Steffen!

Reply:
You can use the function that @steffen-eiden mentions. Otherwise, fixup the style of this function by using an `if let`. 

> that sysfs entry (if it exists) just says if its a SE-guest (not backed by attestation) and can be used as an early exit for e.g. an attestation flow.

This detection function should only be triggered if attestation is supported. Is that what is happening here?

Reply:
I just noticed @steffen-eiden made it public. Yeah, I'll just use it directly.